src/gill/NeckAndShoulderPattern.java

```diff
@@ -0,0 +1,95 @@
+    {
+        super(userName, dataStore, template);
+    }
+
+    @Override
+    protected EPattern assignPattern()
+    {
+        return EPattern.GILL_NECK_SHOULDER;
+    }
+
+    @Override
+    protected void defineRequiredMeasurements() throws Exception
+    {
+        measurements.addMeasurement(new Measurement("shoulderLengthRight", "A11"));
+        measurements.addMeasurement(new Measurement("frontNeckDiagonal", "A73"));
+        measurements.addMeasurement(new Measurement("backNeckDiagonal", "A74"));
+
+        // TODO: Daniel to finish
+        measurements.addMeasurement(new Measurement("NeckBaseWidth", "A93"));
+        measurements.addMeasurement(new Measurement("ShoulderLengthLeft", "A94"));
+        measurements.addMeasurement(new Measurement("LShoulderPointZ", "A95"));
+        measurements.addMeasurement(new Measurement("RShoulderPointZ", "A96"));
+        measurements.addMeasurement(new Measurement("LSideNeckPointZ", "A97"));
+        measurements.addMeasurement(new Measurement("RSideNeckPointZ", "A98"));
+    }
+
+    @Override
+    public void createBlocks()
+    {
+        // Pull from store
+        var shoulderLengthRight = get("shoulderLengthRight");
+        var frontNeckDiagonal = get("frontNeckDiagonal");
+        var backNeckDiagonal = get("backNeckDiagonal");
+
+        // TODO: Daniel to finish parsing measurements into local variables
+        var NeckBaseWidth = get("NeckBaseWidth");
+        var ShoulderLengthLeft = get("ShoulderLengthLeft");
+        var LShoulderPointZ = get("LShoulderPointZ");
+        var RShoulderPointZ = get("RShoulderPointZ");
+        var LSideNeckPointZ = get("LSideNeckPointZ");
+        var RSideNeckPointZ = get("RSideNeckPointZ");
+
+        // Build block
+        Block fullBlock = new Block(userName + "_Gill_NeckShoulder");
+        blocks.add(fullBlock);
+
+        //If conditions
+        double Rsidepoint = ((RSideNeckPointZ < 0 &&  RShoulderPointZ < 0) || (RSideNeckPointZ > 0 &&  RShoulderPointZ < 0)) ? abs(RSideNeckPointZ - RShoulderPointZ) : -abs(RSideNeckPointZ - RShoulderPointZ);
+        double Lsidepoint = ((LSideNeckPointZ < 0 &&  LShoulderPointZ < 0) || (LSideNeckPointZ > 0 &&  LShoulderPointZ < 0)) ? abs(LSideNeckPointZ - LShoulderPointZ) : -abs(LSideNeckPointZ - LShoulderPointZ);
+
+        // Defining each point
+        var point1 = new Vector2D(0, backNeckDiagonal);
+
+        // TODO: Daniel to finish
+        var point2 = new Vector2D(0, -frontNeckDiagonal);
+        var point3 = new Vector2D(-(NeckBaseWidth*0.5), 0);
+        var point4 = new Vector2D(-(NeckBaseWidth*0.5 + shoulderLengthRight), Rsidepoint);
+        var point5 = new Vector2D(NeckBaseWidth*0.5, 0);
+        var point6 = new Vector2D(NeckBaseWidth*0.5 + ShoulderLengthLeft, Lsidepoint);
+
+
+        // All the keypoints/vectors added
+        fullBlock.addKeypoint(point1);
+        fullBlock.addCircularArc(point1, point3, 0.9, true);
```

Comment:
The multipliers in this section seem a bit arbitrary. Can we add them as Arb_<mesurement_name> variables? See line 40 in the Gill Skirt patterns as an example of arbitrary values.

Reply:
Waiting on ScantoPattern on that

Reply:
OK let's leave this for now.